lang/rust/avro/tests/union_schema.rs

```diff
@@ -0,0 +1,307 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use serde::{Deserialize, Serialize};
+use serde::de::DeserializeOwned;
+use apache_avro::{from_value, Schema, Writer, Reader, Codec};
+
+
+static SCHEMA_A_STR: &str = r#"{
+        "name": "A",
+        "type": "record",
+        "fields": [
+            {"name": "field_a", "type": "float"}
+        ]
+    }"#;
+
+static SCHEMA_B_STR: &str = r#"{
+        "name": "B",
+        "type": "record",
+        "fields": [
+            {"name": "field_b", "type": "long"}
+        ]
+    }"#;
+
+static SCHEMA_C_STR: &str = r#"{
+        "name": "C",
+        "type": "record",
+        "fields": [
+            {"name": "field_union", "type": ["A", "B"]},
+            {"name": "field_c", "type": "string"}
+        ]
+    }"#;
+
+#[derive(Serialize,Deserialize,Clone, PartialEq, Debug)]
+struct A {
+    field_a: f32,
+}
+
+#[derive(Serialize,Deserialize,Clone, PartialEq, Debug)]
+struct B {
+    field_b: i64,
+}
+
+#[derive(Serialize,Deserialize,Clone, PartialEq, Debug)]
+#[serde(untagged)]
+enum UnionAB {
+    A(A),
+    B(B),
+}
+
+#[derive(Serialize,Deserialize,Clone, PartialEq, Debug)]
+struct C {
+    field_union: UnionAB,
+    field_c: String
+}
+
+fn encode_decode<T> (input: &T,schema: &Schema,schemata: &Vec<Schema>) -> T
+    where T: DeserializeOwned + Serialize {
+    let mut encoded: Vec<u8> = Vec::new();
+    let mut writer = Writer::with_schemata(&schema, schemata.iter().collect(), &mut encoded, Codec::Null);
+    writer.append_ser(input).unwrap();
+    writer.flush().unwrap();
+
+    let mut reader = Reader::with_schemata(schema, schemata.iter().collect(), encoded.as_slice()).unwrap();
+    from_value::<T>(&reader.next().unwrap().unwrap()).unwrap()
+}
+
+
+#[test]
+fn union_schema_round_trip_no_null()  {
+    let schemata: Vec<Schema> = Schema::parse_list(&[SCHEMA_A_STR, SCHEMA_B_STR, SCHEMA_C_STR]).expect("parsing schemata");
+
+    {
```

Comment:
No need of the scoping. Rust is fine with shadowing.

Reply:
Done - thanks